tests/test_js_files.py

```
@@ -0,0 +1,67 @@
+    assert os.path.isfile(filepath)
+    with open(filepath, 'r', encoding='utf-8') as f:
+        content = f.read()
+
+    assert len(content.strip()) > 0, f"JS file is empty: {filepath}"
+
+    # Basic brackets / parentheses balancing check
+    stack = []
+    pairs = {')': '(', '}': '{', ']': '['}
+
+    # Token-aware cleaning of comments, string literals, template strings, and regex literals
+    token_pattern = r'/\*[\s\S]*?\*/|//.*|"(?:\\.|[^"\\])*"|\'(?:\\.|[^\'\\])*\'|`(?:\\.|[^`\\])*`|/(?![/*])(?:\\/|[^/\n])+/[gimsuy]*'
```

Ruff (0.16.2)

[error] Possible hardcoded password assigned to: "token_pattern"

(S105)